test_conformance/api/test_mem_object_properties_queries.cpp

```diff
@@ -0,0 +1,294 @@
+        }
+        test_error(error, "clCreateImageWithProperties failed");
+    }
+    if (test_case.obj_t == buffer)
+    {
+        if (test_case.query_t == NULL_PATH)
+        {
+            test_object = clCreateBufferWithProperties(
+                context, NULL, flags, local_data.size() * sizeof(cl_uint),
+                local_data.data(), &error);
+        }
+        else
+        {
+            test_object = clCreateBufferWithProperties(
+                context, test_case.properties.data(), flags,
+                local_data.size() * sizeof(cl_uint), local_data.data(), &error);
+        }
+
+        test_error(error, "clCreateBufferWithProperties failed.");
+    }
+    clGetMemObjectInfo(test_object, CL_MEM_PROPERTIES, 0, NULL, &set_size);
+    test_error(error,
+               "clGetMemObjectInfo failed asking for CL_MEM_PROPERTIES.");
+    if (test_case.query_t == NULL_PATH && set_size == 0)
+    {
+        return TEST_PASS;
+    }
+    if (test_case.query_t == NON_NULL_PATH && set_size == 0)
+    {
+        log_error("clGetMemObjectInfo - invalid value returned");
+        return TEST_FAIL;
+    }
+    cl_uint number_of_props = set_size / sizeof(cl_mem_properties);
+    object_properties_check.resize(number_of_props);
+    clGetMemObjectInfo(test_object, CL_MEM_PROPERTIES, set_size,
+                       object_properties_check.data(), NULL);
+    test_error(error,
+               "clGetMemObjectInfo failed asking for CL_MEM_PROPERTIES.");
+
+    if (object_properties_check != test_case.properties)
+    {
+        log_error("clGetMemObjectInfo returned different properties than was "
+                  "set initially.\n");
+        error = TEST_FAIL;
+    }
+    return error;
+}
+
+
+int run_test_query_properties(cl_context context, cl_command_queue queue,
+                              test_data test_case)
+{
+    int error = CL_SUCCESS;
+    log_info("test case description: %s\n", test_case.description.c_str());
+
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+    clMemWrapper obj_src;
+    clMemWrapper obj_dst;
+    clEventWrapper event;
+    MTdata d;
+
+    cl_mem_flags flags;
+    cl_uint size_x;
+    cl_uint size_y;
+    size_t size;
+    size_t global_dim[2];
+
+    size_x = 4;
+    size_y = 4;
+    size = size_x * size_y * 4;
+    global_dim[0] = size_x;
+    global_dim[1] = size_y;
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { size_x, size_y, 1 };
+
+    std::vector<cl_uint> src_data(size);
+    std::vector<cl_uint> dst_data(size);
+    d = init_genrand(gRandomSeed);
+
+    generate_random_data(kUInt, size, d, src_data.data());
+    generate_random_data(kUInt, size, d, dst_data.data());
+    free_mtdata(d);
+    d = NULL;
+    const char* kernel_src = test_case.src.c_str();
+    error = create_single_kernel_helper_with_build_options(
+        context, &program, &kernel, 1, &kernel_src,
+        test_case.kernel_name.c_str(), "-cl-std=CL3.0");
```

Comment:
Minor: this test is testing OpenCL 3.0 APIs, so it's probably safe to pass `-cl-std=CL3.0` here,  But, there's nothing in either kernel that actually requires OpenCL 3.0, so I'd consider omitting any build options to reduce complexity.